schemes.go

```diff
@@ -921,6 +925,71 @@ func (scheme *SchemeManager) parseContents(conf *Configuration) error {
 		conf.Issuers[issuer.Identifier()] = issuer
 		return scheme.parseCredentialsFolder(conf, issuer, filepath.Join(dir, "Issues"))
 	})
+
+	if err != nil {
+		return err
+	}
+
+	// get a map of CredentialTypes of the current scheme
+	allCredTypesForScheme := map[CredentialTypeIdentifier]CredentialType{}
+	for _, credType := range conf.CredentialTypes {
+		if credType.SchemeManagerID == scheme.ID {
+			allCredTypesForScheme[credType.Identifier()] = *credType
+		}
+	}
+
+	// validate that there are no circular dependencies
+	for _, item := range allCredTypesForScheme {
+		if err := item.validateDependencies(conf, []CredentialTypeIdentifier{}); err != nil {
```

Comment:
Instead of having a second loop here, would it not be possible to invoke `validateDependencies()` in the first loop above within the if?

Reply:
yeah, that part was utterly useless. fixed now